maintenance/wikia/sql/dataware-schema.sql

```diff
@@ -14,562 +14,1046 @@
-  KEY `page_latest` (`page_wikia_id`,`page_namespace`,`page_latest`)
-) ENGINE=InnoDB DEFAULT CHARSET=utf8;
-SET character_set_client = @saved_cs_client;
+  KEY `page_title_lower_wikia` (`page_title_lower`,`page_wikia_id`,`page_namespace`),
+  KEY `page_wikia_title_lower` (`page_wikia_id`,`page_title_lower`,`page_namespace`),
+  KEY `page_latest` (`page_wikia_id`,`page_namespace`,`page_latest`),
+  KEY `page_last_indexed` (`page_wikia_id`,`page_last_indexed`,`page_id`),
+  KEY `page_title_lower_edits` (`page_title_lower`,`page_wikia_id`,`page_edits`),
+  KEY `page_wikia_edited` (`page_wikia_id`,`page_last_edited`,`page_title_lower`),
+  KEY `page_namespace_lower` (`page_wikia_id`,`page_namespace`,`page_title_lower`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
+/*!40101 SET character_set_client = @saved_cs_client */;
 
 --
--- Table structure for table `text_regex`
+-- Table structure for table `parser_speed_article`
 --
 
-SET @saved_cs_client     = @@character_set_client;
-SET character_set_client = utf8;
-CREATE TABLE IF NOT EXISTS `text_regex` (
-  `tr_id` int(8) NOT NULL auto_increment,
-  `tr_text` varchar(255) NOT NULL,
-  `tr_timestamp` char(14) NOT NULL,
-  `tr_user` int(8) NOT NULL,
-  `tr_subpage` varchar(255) NOT NULL,
-  PRIMARY KEY  (`tr_id`),
-  UNIQUE KEY `tr_text_subpage` (`tr_text`,`tr_subpage`),
-  KEY `tr_subpage` (`tr_subpage`),
-  KEY `tr_timestamp` (`tr_timestamp`),
-  KEY `tr_user` (`tr_user`)
-) ENGINE=InnoDB AUTO_INCREMENT=202 DEFAULT CHARSET=latin1;
-SET character_set_client = @saved_cs_client;
+DROP TABLE IF EXISTS `parser_speed_article`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `parser_speed_article` (
+  `wiki_id` int(11) NOT NULL,
+  `article_id` int(11) NOT NULL,
+  `page_ns` int(11) DEFAULT NULL,
+  `average_time` decimal(10,3) DEFAULT NULL,
+  `minimum_time` decimal(10,3) DEFAULT NULL,
+  `maximum_time` decimal(10,3) DEFAULT NULL,
+  `wikitext_size` int(11) DEFAULT NULL,
+  `html_size` int(11) DEFAULT NULL,
+  `updated` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  `exp_func_count` int(11) DEFAULT NULL,
+  `node_count` int(11) DEFAULT NULL,
+  `post_expand_size` int(11) DEFAULT NULL,
+  `temp_arg_size` int(11) DEFAULT NULL,
+  PRIMARY KEY (`wiki_id`,`article_id`),
+  KEY `namespace` (`wiki_id`,`page_ns`),
+  KEY `avg_time` (`wiki_id`,`average_time`),
+  KEY `min_time` (`wiki_id`,`minimum_time`),
+  KEY `max_time` (`wiki_id`,`maximum_time`),
+  KEY `wikitext_size` (`wiki_id`,`wikitext_size`),
+  KEY `html_size` (`wiki_id`,`html_size`),
+  KEY `exp_func_count` (`wiki_id`,`exp_func_count`),
+  KEY `node_count` (`wiki_id`,`node_count`),
+  KEY `post_expand_size` (`wiki_id`,`post_expand_size`),
+  KEY `temp_arg_size` (`wiki_id`,`temp_arg_size`)
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
+/*!40101 SET character_set_client = @saved_cs_client */;
```

Comment:
amount of indexes here is frightening, it should be "analytical" table for some stats, inserting data into it is have to be quite expensive with so many indexes